src/common/transformations/src/transformations/common_optimizations/transpose_sinking_binary.cpp

```diff
@@ -0,0 +1,75 @@
+#include "transformations/common_optimizations/transpose_sinking_binary.hpp"
+
+#include <openvino/opsets/opset9.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <transformations/utils/utils.hpp>
+#include <utility>
+
+#include "itt.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/label.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+#include "openvino/util/log.hpp"
+#include "transformations/common_optimizations/transpose_sinking_utils.hpp"
+
+using namespace ov::pass::pattern;
+using namespace ov;
+using namespace ov::opset9;
+using namespace transpose_sinking;
+
+ov::pass::TransposeSinkingBinaryElementwiseForward::TransposeSinkingBinaryElementwiseForward() {
+    MATCHER_SCOPE(TransposeSinkingBinaryElementwiseForward);
+
+    auto main_node_label = wrap_type<op::util::BinaryElementwiseArithmetic>(IfNodeHasTransposeInputs);
+
+    matcher_pass_callback matcher_pass_callback = [=](Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        auto& main_node_output = pattern_to_output.at(main_node_label);
+        auto main_node = main_node_output.get_node_shared_ptr();
+
+        TransposeInputsInfo transpose_input_info = GetFirstTransposeInput(main_node);
+
+        sink_forward::UpdateInputTransposes(main_node, transpose_input_info);
+        for (auto& new_node : sink_forward::InsertOutputTransposes(main_node, transpose_input_info)) {
+            register_new_node(new_node);
+        }
+
+        return true;
+    };
+
+    auto m = std::make_shared<Matcher>(main_node_label, matcher_name);
+    register_matcher(m, matcher_pass_callback);
+}
+
+ov::pass::TransposeSinkingBinaryElementwiseBackward::TransposeSinkingBinaryElementwiseBackward() {
+    MATCHER_SCOPE(TransposeSinkingBinaryElementwiseBackward);
+
+    auto main_node_label = wrap_type<op::util::BinaryElementwiseArithmetic>(consumers_count(1));
+
+    auto transpose_const_label = wrap_type<Constant>(consumers_count(1));
```

Comment:
consumers_count(1) is a quite strict restriction, could you create jira story to resolve it?

Reply:
created
https://jira.devtools.intel.com/browse/CVS-96628